pydoctor/model.py

```diff
@@ -656,22 +656,21 @@ def addPackage(self, dirpath, parentPackage=None):
                 self.addModuleFromPath(package, fullname)
 
     def addModuleFromPath(self, package, path):
-        for (suffix, mode, impl) in imp.get_suffixes():
+        for suffix in importlib.machinery.all_suffixes():
             if not path.endswith(suffix):
                 continue
             module_name = os.path.basename(path[:-len(suffix)])
-            if impl == imp.C_EXTENSION:
+            if suffix in importlib.machinery.EXTENSION_SUFFIXES:
                 if not self.options.introspect_c_modules:
                     continue
                 if package is not None:
                     module_full_name = f'{package.fullName()}.{module_name}'
                 else:
                     module_full_name = module_name
-                py_mod = imp.load_module(
-                    module_full_name, open(path, 'rb'), path,
-                    (suffix, mode, impl))
+                py_mod = importlib.util.spec_from_file_location(
+                    module_full_name, path)
```

Comment:
The problem is that we don't have automated tests for this branch

Reply:
To test the full flow would require using a C extension, which is a pain to set up. However, we could move the import handling into `introspectModule()`, which can also be used on pure Python and is in fact unit tested that way. Then adding type annotations will let mypy verify the bit of code that remains without unit test coverage.